File: app/src/main/java/com/worldventures/dreamtrips/social/ui/friends/model/converter/MutualsConverter.java
package com.worldventures.dreamtrips.social.ui.friends.model.converter;

import com.worldventures.core.converter.Converter;
import com.worldventures.core.model.User;
import com.worldventures.dreamtrips.api.session.model.MutualFriends;

import io.techery.mappery.MapperyContext;

public class MutualsConverter implements Converter<MutualFriends, User.MutualFriends> {
   @Override
   public Class<MutualFriends> sourceClass() {
      return MutualFriends.class;
   }

   @Override
   public Class<User.MutualFriends> targetClass() {
      return User.MutualFriends.class;
   }

   @Override
   public User.MutualFriends convert(MapperyContext mapperyContext, MutualFriends mutualFriends) {
      return new User.MutualFriends(mutualFriends.count());
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/friends/model/converter/MutualsConverter.kt
package com.worldventures.dreamtrips.social.service.friends.model.converter

import com.worldventures.core.converter.Converter
import com.worldventures.core.model.User
import com.worldventures.dreamtrips.api.session.model.MutualFriends

import io.techery.mappery.MapperyContext

class MutualsConverter : Converter<MutualFriends, User.MutualFriends> {

   override fun sourceClass(): Class<MutualFriends> = MutualFriends::class.java

   override fun targetClass(): Class<User.MutualFriends> = User.MutualFriends::class.java

   override fun convert(context: MapperyContext, source: MutualFriends)
         : User.MutualFriends = User.MutualFriends(source.count())
}
